libs/designsystem/calendar/src/calendar.component.ts

```diff
@@ -479,13 +466,14 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
   private changeActiveView(index: number, unit: TimeUnit) {
     if (index === 0) return;
     this.activeMonth = add(this.activeMonth, { [unit]: index });
+    this.focussedDate = add(this.focussedDate, { [unit]: index });
 
     this.refreshActiveMonth();
   }
 
   get _canNavigateBack(): boolean {
-    const reachedPastDatesLimit =
-      this.disablePastDates && isSameMonth(this.activeMonth, this.todayDate);
+    const today = this.todayDate ? startOfDay(this.todayDate) : startOfDay(new Date());
```

Comment:
Consider simplifying:
```suggestion
    const today = startOfDay(this.todayDate ?? new Date());
```